pkg/apis/core/validation/cloudprofile.go

```diff
@@ -97,6 +97,12 @@ func validateCloudProfileKubernetesSettings(kubernetes core.KubernetesSettings,
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("versions[]").Child("expirationDate"), latestKubernetesVersion.ExpirationDate, fmt.Sprintf("expiration date of latest kubernetes version ('%s') must not be set", latestKubernetesVersion.Version)))
 	}
 
+	for _, lifecycle := range latestKubernetesVersion.Lifecycle {
+		if lifecycle.Classification != core.ClassificationExpired {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("versions[]").Child("lifecycle"), latestKubernetesVersion.ExpirationDate, fmt.Sprintf("expiration date of latest kubernetes version ('%s') must not be set", latestKubernetesVersion.Version)))
+		}
+	}
```

Comment:
I think the `!=` condition is wrong here because this will make the versions to be `expired`. 😅

Also it is not required to loop because the order is already enforced.

```suggestion
	if len(latestKubernetesVersion.Lifecycle) > 0 {
		if latestKubernetesVersion.Lifecycle[len(latestKubernetesVersion.Lifecycle)-1].Classification == core.ClassificationExpired {
			allErrs = append(allErrs, field.Invalid(fldPath.Child("versions[]").Child("lifecycle"), latestKubernetesVersion.ExpirationDate, fmt.Sprintf("expiration date of latest kubernetes version ('%s') must not be set", latestKubernetesVersion.Version)))
		}
	}
```